src/http/validations/device.js

```diff
@@ -13,6 +13,19 @@ const getSingleDevice = (req, res, next) => {
   return next();
 };
 
-export default {
-  getSingleDevice,
+const updateDevice = (req, res, next) => {
+  const { deviceId } = req.params;
+  const { label } = req.body;
+
+  const fieldErrors = new FieldErrors();
+
+  if (!deviceId || typeof (deviceId) !== 'string') fieldErrors.addError('deviceId', 'deviceId is a required string');
```

Comment:
Actually, you can(and should) validate that `deviceId` is a valid resource Id. Since we are making use of mongoDB Object Id as our resource IDs, we should validate that the given `deviceId` is a valid ObjectId. I already wrote a method in the db module to handle this, you can call:
```
db.validResourceId(deviceId)
```
to check if the `deviceId` is valid

Reply:
fixed